x-pack/plugins/apm/server/lib/services/get_services/get_service_transaction_stats.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { AgentName } from '../../../../typings/es_schemas/ui/fields/agent';
+import { EventOutcome } from '../../../../common/event_outcome';
+import {
+  AGENT_NAME,
+  EVENT_OUTCOME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+} from '../../../../common/elasticsearch_fieldnames';
+import { rangeFilter } from '../../../../common/utils/range_filter';
+import {
+  getDocumentTypeFilterForAggregatedTransactions,
+  getProcessorEventForAggregatedTransactions,
+  getTransactionDurationFieldForAggregatedTransactions,
+} from '../../helpers/aggregated_transactions';
+import { getBucketSize } from '../../helpers/get_bucket_size';
+import { ServicesItemsSetup } from './get_services_items';
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  searchAggregatedTransactions: boolean;
+}
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+export async function getServiceTransactionStats({
+  setup,
+  searchAggregatedTransactions,
+}: AggregationParams) {
+  const { apmEventClient, start, end, esFilter } = setup;
+
+  const count = {
+    value_count: {
+      field: getTransactionDurationFieldForAggregatedTransactions(
+        searchAggregatedTransactions
+      ),
+    },
+  };
+
+  const metrics = {
+    count,
+    avg_duration: {
+      avg: {
+        field: getTransactionDurationFieldForAggregatedTransactions(
+          searchAggregatedTransactions
+        ),
+      },
+    },
+    failure: {
+      filter: { term: { [EVENT_OUTCOME]: EventOutcome.failure } },
+      aggs: { count },
+    },
+    successful: {
+      filter: { term: { [EVENT_OUTCOME]: EventOutcome.success } },
+      aggs: { count },
+    },
```

Comment:
Can you try comparing the performance of two `filter` aggregations with a `terms` aggregation with the `filter` option set? See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#_filtering_values_4.

Reply:
@dgieselaar I profiled both queries and using `terms aggregation` with `include` property the query runs ~300ms faster than the having two `filter aggregations`.  WDYT? is it worth changing it on this PR?

Reply:
sounds good! but 300ms relative to what?

Reply:
I ran both queries on the `Search Profiler` with a time range of 15 days on `apm-*` indices. Is it what you mean?

Reply:
What I mean was, what is the total time of both queries? e.g., it can be 300ms faster than 301ms, which is an incredible improvement, or it can be 300ms faster than 30 minutes, which is a negligible improvement.

Reply:
The query with `filter aggregation` took `~2.344ms` and the one with `terms aggregation with include` took `~2.003`.

Reply:
@dgieselaar I replaced it to use the same aggregation that is used on transaction error rate https://github.com/elastic/kibana/pull/86434/commits/9fe3789e28d4904a2d410559429f844ecd0fafad#diff-b83c835e7e4abd7324ea52a0f588332a6934edde706f526f6ee5c571e5757b80R62